src/forward_index.c

```diff
@@ -238,15 +246,15 @@ int forwardIndexTokenFunc(void *ctx, const Token *tokInfo) {
       for (int i = 0; i < array_len(t_data->ids); ++i) {
         synonym_len = SynonymMap_IdToStr(t_data->ids[i], synonym_buff, SYNONYM_BUFF_LEN);
         ForwardIndex_HandleToken(tokCtx->idx, synonym_buff, synonym_len, tokInfo->pos,
-                                 tokCtx->fieldScore, tokCtx->fieldId, 0, 1, true);
+                                 tokCtx->fieldScore, tokCtx->fieldId, TOKOPT_F_COPYSTR);
       }
     }
   }
 
   if (tokInfo->phoneticsPrimary) {
     ForwardIndex_HandleToken(tokCtx->idx, tokInfo->phoneticsPrimary,
                              strlen(tokInfo->phoneticsPrimary), tokInfo->pos, tokCtx->fieldScore,
-                             tokCtx->fieldId, 0, 0, true);
+                             tokCtx->fieldId, TOKOPT_F_COPYSTR);
```

Comment:
You are passing the same options here and above but it was not the same before the commit ...

Reply:
The change is intentional. Without the change there is a memory leak (it makes more sense if you see the individual commit). Basically the commit frees the string at the source, and also makes sure that the string here is copied.